Assets/Scripts/NetworkClient.cs

```diff
@@ -34,66 +36,116 @@ void Start()
     {
         _player = GetComponent<GfxReplayPlayer>();
         Assert.IsTrue(_player);  // our object should have a GfxReplayPlayer
+        _configLoader = GetComponent<ConfigLoader>();
+        Assert.IsTrue(_configLoader);
 
         _xrInputHelper = new XRInputHelper();
 
-        bool isHttps = false;
-        wsProtocol = isHttps ? "wss" : "ws";
-        _fullServerAddress = $"{wsProtocol}://{serverAddress}:{serverPort}";
-
-        SetConnectionState(false);
-
-        ConnectWebSocket();
+        string[] serverLocations = _configLoader.AppConfig.serverLocations;
+        Assert.IsTrue(serverLocations.Length > 0);
+        foreach (string location in serverLocations)
+        {
+            string adjustedLocation = location;
+            if (!adjustedLocation.Contains(":"))
+            {
+                adjustedLocation += ":" + defaultServerPort;
+            }
+
+            bool isHttps = false;
+            string wsProtocol = isHttps ? "wss" : "ws";
+            _serverURLs.Add($"{wsProtocol}://{adjustedLocation}");
+        }
 
-        // Attempt reconnection every X seconds
-        InvokeRepeating("AttemptReconnection", 0.0f, 5.0f);
+        StartCoroutine(TryConnectToServers());
 
         // Keep sending messages at every 0.1s
         InvokeRepeating("SendClientState", 0.0f, 0.1f);
     }
 
     void Update()
     {
-        if (websocket != null)
+        if (mainWebSocket != null)
         {
 #if !UNITY_WEBGL || UNITY_EDITOR
-            websocket.DispatchMessageQueue();
+            mainWebSocket.DispatchMessageQueue();
 #endif
         }
     }
 
-    async void ConnectWebSocket()
+    private IEnumerator TryConnectToServers()
     {
-        websocket = new WebSocket(_fullServerAddress);
+        while (true)
+        {
+            if (mainWebSocket == null)
+            {
+                yield return new WaitForSeconds(1); // always wait 1s before next try
+
+                currentServerIndex++;
+                if (currentServerIndex >= _serverURLs.Count)
+                {
+                    currentServerIndex = 0; // Reset to try again from the beginning.
+                }
+
+                string url = _serverURLs[currentServerIndex];
+                Debug.Log("Attempting to connect to: " + url);
+
+                var websocket = new WebSocket(url);
+                var connectTask = ConnectWebSocket(websocket, url);
+
+                // Wait for 8s, then cancel the connection attempt (i.e. time out)
+                yield return new WaitForSeconds(8);
```

Comment:
What happens if the game is closed at this point - i.e. before the engine fires the portion after this scheduled `yield`? Does the library correctly release the socket?

Reply:
I don't know. I guess the thing to test and look out for will be a zombie connection to the server. I haven't seen this yet in my testing.